### PR TITLE
Implement handle, Ping handler

### LINKED ISSUE #72: [MessageHandler] implements handle, Ping handler
implements MessageHandler handle, more specifically `Ping` message handler

### PR DESCRIPTION
**Pull request by AgwaB**
resolved: #72 

details: 

-Implement `New` function more detail
-pb message is not implemented exactly yet so implement ping handler simply

### CODE REVIEW

File: swim.go [hunk 1]
```diff
@@ -182,7 +214,14 @@ func (s *SWIM) handle(msg pb.Message) {
 
 	switch msg.Payload.(type) {
 	case *pb.Message_Ping:
-		// handle ping
+		s.messageEndpoint.Send(msg.Address, pb.Message{
+			Seq: "",
+			Address: "",
+			Payload: &pb.Message_Ack{
```
Comment (junbeomlee) [inline swim.go]: You must put your address(need to add member(me) into swim struct) in the address value.
Seq will be uuid. I think we should change the name of seq to id?

File: swim.go [hunk 2]
```diff
@@ -182,7 +214,14 @@ func (s *SWIM) handle(msg pb.Message) {
...
+				Ack: &pb.Ack{Payload: ""},
+			},
+			PiggyBack: &pb.PiggyBack{},
+		})
```
Comment (junbeomlee) [inline swim.go]: get piggyback from pbkstore if it is not empty

File: swim.go (hunk 1 above)
Comment (hihiboss) [reply swim.go]: I agree to change the name of seq to id.

File: swim.go (hunk 2 above)
Comment (AgwaB) [reply swim.go]: pbkstore push and get []byte type. I don't understand why pbkstore use []byte type.  Does PiggyBack convey member information(ex> address, seq(id), incarnation number)?

File: swim.go (hunk 1 above)
Comment (zeroFruit) [reply swim.go]: +1

File: swim.go [hunk 3]
```diff
@@ -182,7 +219,15 @@ func (s *SWIM) handle(msg pb.Message) {
 
 	switch msg.Payload.(type) {
 	case *pb.Message_Ping:
-		// handle ping
+		Address := s.config.BindAddress + ":" + string(s.config.BindPort)
```
Comment (zeroFruit) [inline swim.go]: how about refactor to function this logic `handleXXX(msg pb.Message)` ?

File: swim.go [hunk 4]
```diff
@@ -45,21 +52,51 @@ type SWIM struct {
...
+
+	messageEndpoint := setMessageEndpoint(config, messageEndpointConfig, &swim, awareness)
+	swim.messageEndpoint = messageEndpoint
+
+	return &swim
+}
+
+// set MessageEndpoint
+func setMessageEndpoint(config *Config, messageEndpointConfig MessageEndpointConfig, messageHandler MessageHandler, awareness *Awareness) *MessageEndpoint {
+	packetTransportConfig := PacketTransportConfig{
```
Comment (zeroFruit) [inline swim.go:86]: why don't you just integrate this function into `New` or I think this function name would be better like `messageEndpointFactory`. I think this function not set MessageEndpoint to swim just create and return

File: swim.go (hunk 3 above)
Comment (junbeomlee) [reply swim.go]: I agree

File: swim.go (hunk 4 above)
Comment (AgwaB) [reply swim.go:86]: you mean, you think that it is better to implement into `New` function at once?

Comment (zeroFruit) [reply swim.go:86]: yes my original intention was implementing once but swim is used for creating `MessageEndpoint`.. then I think it is better to change func name

**Comment by AgwaB** [comment]
I make a mistake in 'commit' so close it and PR it again.

**Pull request closed without merging**